test/index.test.ts

```diff
@@ -3,10 +3,20 @@ import { expect, test, vi } from 'vitest'
 
 import { Bench, hrtimeNow, now, type Task } from '../src'
 
+/**
+ * @param ms amount of time to sleep in milliseconds
+ */
+function sleep (ms: number): void {
+  const start = performance.now()
+  while (performance.now() - start < ms) {
+    // noop
+  }
```

Comment:
That `sleep` implementation will hog the event loop. Do you need the timing precision brought here versus a promise based `sleep`?

Reply:
We can't have a promise based sleep here precisely because the tests are synchronous. We have to either spend the time in the tests or mock timers for the same reason.